### PR TITLE
Daemonset update and bug fix

### PR DESCRIPTION
**Pull request by rickkkkc137**

### CODE REVIEW

File: cmd/daemonset-check/input.go [hunk 1]
```diff
@@ -85,4 +86,55 @@ func parseInputValues() {
...
+				if len(splitkv) > 1 {
+					findte := strings.Split(splitkv[1], ":")
+					//does toleration have an effect
+					if len(findte) > 1 {
+						//get value/effect and generate toleration
+						tvalue := findte[0]
+						teffect := findte[1]
+						t := corev1.Toleration{
+							Key: splitkv[0],
+							Operator: "Equal",
```
Comment (jonnydawg) [inline cmd/daemonset-check/input.go:106]: try using `corev1.TolerationOpEqual`

File: cmd/daemonset-check/input.go [hunk 2]
```diff
@@ -85,4 +86,55 @@ func parseInputValues() {
...
+							Value: tvalue,
+							Effect: corev1.TaintEffect(teffect),
+						}
+						tolerations = append(tolerations, t)
+					} else {
+						// generate based on splitkv
+						t := corev1.Toleration{
+							Key: splitkv[0],
+							Operator: "Equal",
+							Value: splitkv[1],
```
Comment (jonnydawg) [inline cmd/daemonset-check/input.go:116]: same here

File: cmd/daemonset-check/input.go [hunk 3]
```diff
@@ -85,4 +86,55 @@ func parseInputValues() {
...
+						}
+						tolerations = append(tolerations, t)
+					}
+				} else {
+					t := corev1.Toleration{
+						Key: toleration,
+						Operator: "Exists",
```
Comment (jonnydawg) [inline cmd/daemonset-check/input.go:123]: there's a `corev1.TolerationOpExists` as well

File: cmd/daemonset-check/input.go [hunk 4]
```diff
@@ -85,4 +86,55 @@ func parseInputValues() {
...
+					}
+					tolerations = append(tolerations, t)
+				}
+			}
+		} else {
+			//generate toleration based on single string value
+			t := corev1.Toleration{
+				Key: tolerationsEnv,
+				Operator: "Exists",
```
Comment (jonnydawg) [inline cmd/daemonset-check/input.go:132]: and same here as above

File: cmd/daemonset-check/input.go (hunk 1 above)
Comment (rjacks161) [reply cmd/daemonset-check/input.go]: Added these

**Review by jonnydawg — APPROVED** [review]
Looks good! Thanks for doing this!

**Comment by jonnydawg** [comment]
Make sure to change the tag value for the daemonset-check in the makefile

**Comment by integrii** [comment]
re-running jobs with @jonnydawg's fix merged

File: cmd/daemonset-check/daemonset-check.yaml [hunk 5]
```diff
@@ -13,6 +13,8 @@ spec:
       - env:
           - name: POD_NAMESPACE
             value: "kuberhealthy"
+          - name: TOLERATIONS
+            value: "kubernetes.io/hostname=test"
```
Comment (integrii) [inline cmd/daemonset-check/daemonset-check.yaml:17]: TODO: add node selector example and comment out both env vars by default

**Review by integrii — CHANGES_REQUESTED** [review]

**Review by integrii — APPROVED** [review]

File: cmd/daemonset-check/create_toleration_test.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"testing"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestCreateToleration(test *testing.T) {
+	//string_tol := ""
+	//string_tol := "blah"
```
Comment (jonnydawg) [inline cmd/daemonset-check/create_toleration_test.go:10]: could we remove these comments?

File: cmd/daemonset-check/input.go [hunk 7]
```diff
@@ -1,13 +1,72 @@
...
 
 import (
+	"errors"
 	"strings"
 	"time"
 
 	kh "github.com/Comcast/kuberhealthy/v2/pkg/checks/external/checkclient"
 	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
 )
```
Comment (jonnydawg) [inline cmd/daemonset-check/input.go:12]: it'd be nice to have comments for these functions :)

**Review by jonnydawg — COMMENTED** [review]
small changes requested, but looks good!

**Review by jonnydawg — APPROVED** [review]

**Comment by jonnydawg** [comment]
👍

**Pull request merged**